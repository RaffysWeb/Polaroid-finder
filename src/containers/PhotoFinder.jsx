import React, { useState, useRef, useEffect } from 'react';

import TextInput from '../components/TextInput';
import Polaroid from '../components/Polaroid';
import { fetchImages } from '../utils/apiRequests';
import useDebounce from '../utils/customHooks';

const PhotoFinder = () => {
  const [state, setState] = useState({ fetchedImages: null });
  const { searchInput, fetchedImages } = state;

  const inputEl = useRef(null);

  const debouncedSearchValue = useDebounce(searchInput, 500);

  useEffect(() => {
    inputEl.current.focus();
  }, []);

  useEffect(() => {
    if (debouncedSearchValue) {
      setState(oldState => ({ ...oldState }));
      fetchImages(debouncedSearchValue).then(results => {
        setState(oldState => ({
          ...oldState,

          fetchedImages: results
        }));
      });
    } else {
      setState(oldState => ({
        ...oldState,
        fetchedImages: null
      }));
    }
  }, [debouncedSearchValue]);

  const onChange = e => {
    setState({ ...state, [e.target.name]: e.target.value });
  };
  console.log(fetchedImages);
  return (
    <>
      <header>
        <h1>Polaroid Finder</h1>
      </header>

      <TextInput
        forwardedRef={inputEl}
        name="searchInput"
        value={searchInput}
        placeholder="Cute dogs..."
        onChange={onChange}
      />
      {!searchInput && <h3>Start typing to search...</h3>}
      {fetchedImages && (
        <section className="polaroids-container">
          {fetchedImages.length ? (
            fetchedImages.map(item => <Polaroid item={item} />)
          ) : (
            <h3>Couldn't find any images please try something else...</h3>
          )}
        </section>
      )}
    </>
  );
};

export default PhotoFinder;
