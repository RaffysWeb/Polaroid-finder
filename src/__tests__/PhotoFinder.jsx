import React from 'react';
import { create } from 'react-test-renderer';
import PhotoFinder from '../containers/PhotoFinder';

describe('PhotoFinder tests', () => {
  test('PhotoFinder it matches the snapshot', () => {
    const component = create(<PhotoFinder />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
