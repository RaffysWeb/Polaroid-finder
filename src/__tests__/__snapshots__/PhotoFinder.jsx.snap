// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoFinder tests PhotoFinder it matches the snapshot 1`] = `
<section
  className="sc-bxivhb ijobHS"
>
  <header>
    <img
      alt=""
      src="/images/polaroid.svg"
    />
    <h1>
      Polaroid Finder
    </h1>
  </header>
  <input
    className="sc-bdVaJa kTNRlN"
    name="searchInput"
    onChange={[Function]}
    placeholder="Cute dogs..."
    type="text"
    value=""
  />
  <h3>
    Start typing to search...
  </h3>
</section>
`;
